File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};
        int total = brown + yellow;
        for(int height = 1; height < total /2 + 1; height++){
            if(total % height == 0){
                int width = total / height;
                if((2 * width ) + (2 * height) - 4 == brown){
                    return new int[]{width, height};
                }
            }
        }
        return answer;
       
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    total = brown + yellow
    for width in range(3, int(total**0.5) + 1):
        if total % width == 0:
            height = total // width
            if 2 * (width + height) - 4 == brown:
                return [height, width]  # width ≥ height로 반환
